test/test_tensordict.py

```diff
@@ -1612,6 +1612,204 @@ def test_repr(self, td_name, device):
+        a: Tensor(torch.Size([1, 4, 3, 2, 1, 5]), dtype={dtype})}},
+    batch_size=torch.Size([1, 4, 3, 2, 1]),
+    device={str(device)},
+    is_shared=False)"""
+        else:
+            expected = f"""TensorDict(
+    fields={{
+        a: Tensor(torch.Size([4, 2, 1, 5]), dtype={dtype})}},
+    batch_size=torch.Size([4, 2, 1]),
+    device={str(device)},
+    is_shared=False)"""
+
+        assert repr(tensordict) == expected
+
+    def test_repr_indexed_nested(self, device, dtype):
+        tensordict = TensorDict({}, [4, 3, 2, 1], device=device)
+        tensordict.set("nested_td", self.nested_td(device))
+
+        expected = """TensorDict(
+    fields={
+        nested_td: TensorDict(
+            fields={
+                b: Tensor(torch.Size([4, 3, 2, 1, 5]), dtype=torch.float32),
+                my_nested_td: TensorDict(
+                    fields={
+                        a: Tensor(torch.Size([4, 3, 2, 1, 5]), dtype=torch.float32)},
+                    batch_size=torch.Size([4, 3, 2, 1]),
+                    device=cpu,
+                    is_shared=False)},
+            batch_size=torch.Size([4, 3, 2, 1]),
+            device=cpu,
+            is_shared=False)},
+    batch_size=torch.Size([4, 3, 2, 1]),
+    device=cpu,
+    is_shared=False)"""
+        assert repr(tensordict) == expected
+
+    def test_repr_indexed_integer(self, device, dtype):
+        tensordict = TensorDict({}, [5], device=device)
+        tensordict.set("k_int", torch.randint(10, (5, 4, 3)))
+
+        expected = """TensorDict(
+    fields={
+        a: Tensor(torch.Size([5, 4, 3]), dtype=torch.float32),
+        k_int: Tensor(torch.Size([5, 4, 3]), dtype=torch.int64)},
+    batch_size=torch.Size([5]),
+    device=cpu,
+    is_shared=False)"""
+        assert repr(tensordict), expected
+
+    def test_repr_indexed_mask(self, device, dtype):
+        tensordict = TensorDict({}, [2], device=device)
+        tensordict.set("a", torch.randn(2, 3))
+        mask = torch.BoolTensor([[1, 0, 1], [1, 0, 1]])
+        masked_td = tensordict[mask]
+        expected = """TensorDict(
+    fields={
+        a: Tensor(torch.Size([4, 1]), dtype=torch.float32)},
+    batch_size=torch.Size([4]),
+    device=cpu,
+    is_shared=False)"""
+        assert repr(masked_td) == expected
+
+    def test_repr_indexed_stack(self, device, dtype):
+        tensordict = TensorDict({}, [5], device=device)
+        td3 = TensorDict({"d": torch.randn(5, 4, 3)}, [5], device=device)
+        stacked_td = stack_td([tensordict, td3], 2)
+
+        expected = """LazyStackedTensorDict(
+    fields={
+    },
+    batch_size=torch.Size([5, 2]),
+    device=cpu,
+    is_shared=False)"""
+        assert repr(stacked_td), expected
+
+    @pytest.mark.skipif(not torch.cuda.device_count(), reason="no cuda")
+    def test_repr_device_to_device(self, device, dtype):
+        dev2 = torch.device(0)
+        td = self.td(device)
+        td2 = td.to(dev2)
+        assert repr(td) == repr(td2)
```

Comment:
If changing the device of a tensordict has no impact on his repr we have a serious problem! 
Have a look at the tests above where the device is used in the formatted string